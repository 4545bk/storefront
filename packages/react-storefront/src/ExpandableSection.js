/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import Typography from '@material-ui/core/Typography'
import withStyles from '@material-ui/core/styles/withStyles'
import AmpExpandableSection from './amp/AmpExpandableSection'
import { inject } from 'mobx-react';
import withTheme from '@material-ui/core/styles/withTheme'

export const styles = (theme) => ({
  root: {
    boxShadow: 'none',
    borderBottom: `1px solid ${theme.palette.divider}`,
    margin: 0,
    paddingLeft: '20px',

    '&::before': {
      display: 'none'
    },

    '& > *:first-child': {
      padding: '0',
      minHeight: '0'
    }
  },

  caption: {
    transition: 'opacity .2s linear'
  },

  expandedCaption: {
    opacity: 0
  },

  largeTitle: {
    fontSize: '18px',
    fontWeight: 'bold',
    color: '#444'
  },

  details: {
    paddingLeft: '0',
    paddingRight: '0'
  },

  summary: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginRight: '25px !important'
  },
  

  withCollapseIcon: {
    transform: 'translateY(-50%) rotate(0deg) !important'
  },

  expandIcon: {},
  collapseIcon: {}
});

@withStyles(styles, { name: 'RSFExpandableSection' })
@withTheme()
@inject(({ app }) => ({ amp: app.amp }))
export default class ExpandableSection extends Component {

  static propTypes = {
    /**
     * The title for the header of the expandable section
     */
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

    /**
     * Text to display to the right of the heading
     */
    caption: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

    /**
     * The icon to use for collapsed groups
     */
    ExpandIcon: PropTypes.func,

    /**
     * The icon to use for expanded groups
     */
    CollapseIcon: PropTypes.func
  }

  constructor({ expanded=false, ExpandIcon, CollapseIcon, theme }) {
    super()

    this.ExpandIcon = ExpandIcon || theme.ExpandIcon || ExpandMoreIcon
    this.CollapseIcon = CollapseIcon || theme.CollapseIcon || this.ExpandIcon
  
    this.state = {	
      expanded	
    }
  }

  render() {
    let { amp, classes, theme, children = [], title, caption, expanded, ExpandIcon: ei, CollapseIcon: ci, ...others } = this.props

    const { ExpandIcon, CollapseIcon } = this

    if (amp) {
      return <AmpExpandableSection ExpandIcon={ExpandIcon} CollapseIcon={CollapseIcon} title={title}>{children}</AmpExpandableSection>
    } else {
      return (
        <ExpansionPanel classes={{ root: classes.root }} expanded={expanded} {...others} onChange={this.onChange}>
          <ExpansionPanelSummary 
            expandIcon={this.state.expanded ? <CollapseIcon className={classes.collapseIcon}/> : <ExpandIcon className={classes.expandIcon}/>} 
            classes={this.getSummaryClasses()}
            >
            <Typography variant="subheading">{title}</Typography>
            { caption && (
              <Typography 
                variant="caption" 
                className={classnames({
                  [classes.caption]: true,
                  [classes.expandedCaption]: expanded
                })}
              >
                {caption}
              </Typography> 
            )}
          </ExpansionPanelSummary>
          <ExpansionPanelDetails classes={{ root: classes.details }}>
            {children}
          </ExpansionPanelDetails>
        </ExpansionPanel>
      )
    }

  }

  componentDidUpdate(prevProps) {
    const { expanded } = this.props
    if (prevProps.expanded !== expanded) {
      this.setState({ expanded })
    }
  }

  /**
   * Gets the classes for the ExpansionPanelSummary
   * Here we add a class to remove the rotate transform if we're using a 
   * separate icon for the collapse state.
   */
  getSummaryClasses() {
    const { classes } = this.props

    const result = {
      content: classes.summary
    }

    if (this.CollapseIcon !== this.ExpandIcon) {
      result.expandIcon = classes.withCollapseIcon
    }

    return result
  }

  onChange = (e, expanded) => {
    if (this.props.onChange) {
      this.props.onChange(e, expanded)
    }
    this.setState({ expanded })
  }

}