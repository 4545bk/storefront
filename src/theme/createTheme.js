import { createMuiTheme, adaptV4Theme } from '@mui/material/styles';
import merge from 'lodash/merge'

/**
 * Creates the default theme for your React Storefront app.  See Material UI's theme documentation
 * for more info: https://mui.com/customization/default-theme/
 * @param {Object} values
 * @return {Object} A material UI theme
 */
export default function createTheme(values = {}) {
  const theme = createMuiTheme(
    adaptV4Theme(merge(
      {},
      {
        zIndex: {
          modal: 999,
          amp: {
            modal: 2147483646,
          },
        },
        headerHeight: 64,
        loadMaskOffsetTop: 64 + 56 + 4,
        drawerWidth: 330,
        overrides: {},
      },
      values,
    )),
  )

  if (!theme.maxWidth) {
    theme.maxWidth = `${theme.breakpoints.values.lg}px`
  }

  return theme
}
