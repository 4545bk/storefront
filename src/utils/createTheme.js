import { createMuiTheme } from '@material-ui/core/styles'
import merge from 'lodash/merge'

export default function createTheme(values = {}) {
  let config = {}

  merge(
    config,
    {
      zIndex: {
        modal: 999,
        amp: {
          modal: 2147483646,
        },
      },
      headerHeight: 64,
      loadMaskOffsetTop: 64 + 56 + 4,
      drawerWidth: 330,
      overrides: {},
    },
    values,
  )

  const theme = createMuiTheme(config)

  if (!theme.maxWidth) {
    theme.maxWidth = `${theme.breakpoints.values.lg}px`
  }

  return theme
}
