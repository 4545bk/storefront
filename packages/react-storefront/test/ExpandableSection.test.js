/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import ExpandableSection from '../src/ExpandableSection'
import createTheme from '../src/createTheme'
import TestProvider from './TestProvider'
import AppModelBase from '../src/model/AppModelBase'

describe('ExpandableSection', () => {

  let theme

  beforeEach(() => {
    theme = createTheme()
  })

  it('should render', () => {
    const app = AppModelBase.create({ amp: false })

    expect(
      mount(
        <TestProvider app={app}>
          <ExpandableSection title="Title">Foo</ExpandableSection>
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render AmpExpandableSection when amp == true', () => {
    const app = AppModelBase.create({ amp: true })

    expect(
      mount(
        <TestProvider app={app}>
          <ExpandableSection title="Title">Foo</ExpandableSection>
        </TestProvider>
      )
    ).toMatchSnapshot()
  })
})