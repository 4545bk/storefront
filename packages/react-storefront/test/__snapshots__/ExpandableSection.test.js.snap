// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableSection should render 1`] = `
<Provider app={{...}}>
  <MuiThemeProvider theme={{...}}>
    <WithStyles(WithTheme(inject-ExpandableSection)) title="Title">
      <WithTheme(inject-ExpandableSection) classes={{...}} title="Title">
        <inject-ExpandableSection theme={{...}} classes={{...}} title="Title">
          <ExpandableSection theme={{...}} classes={{...}} title="Title" amp={false}>
            <WithStyles(ExpansionPanel) classes={{...}} expanded={[undefined]} onChange={[Function]}>
              <ExpansionPanel classes={{...}} expanded={[undefined]} onChange={[Function]} defaultExpanded={false} disabled={false}>
                <WithStyles(Paper) className="MuiExpansionPanel-root-10 RSFExpandableSection-root-1" elevation={1} square={true}>
                  <Paper classes={{...}} className="MuiExpansionPanel-root-10 RSFExpandableSection-root-1" elevation={1} square={true} component="div">
                    <div className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiExpansionPanel-root-10 RSFExpandableSection-root-1">
                      <WithStyles(ExpansionPanelSummary) expandIcon={{...}} classes={{...}} disabled={false} expanded={false} onChange={[Function: value]}>
                        <ExpansionPanelSummary classes={{...}} expandIcon={{...}} disabled={false} expanded={false} onChange={[Function: value]}>
                          <WithStyles(ButtonBase) focusRipple={false} disableRipple={true} disabled={false} component="div" aria-expanded={false} className="MuiExpansionPanelSummary-root-40" onFocusVisible={[Function: value]} onBlur={[Function: value]} onClick={[Function: value]}>
                            <ButtonBase classes={{...}} focusRipple={false} disableRipple={true} disabled={false} component="div" aria-expanded={false} className="MuiExpansionPanelSummary-root-40" onFocusVisible={[Function: value]} onBlur={[Function: value]} onClick={[Function: value]} centerRipple={false} tabIndex="0" type="button">
                              <div onBlur={[Function: handleEvent]} onFocus={[Function: value]} onKeyDown={[Function: value]} onKeyUp={[Function: value]} onMouseDown={[Function: handleEvent]} onMouseLeave={[Function: handleEvent]} onMouseUp={[Function: handleEvent]} onTouchEnd={[Function: handleEvent]} onTouchMove={[Function: handleEvent]} onTouchStart={[Function: handleEvent]} tabIndex="0" className="MuiButtonBase-root-46 MuiExpansionPanelSummary-root-40" role="button" aria-expanded={false} onClick={[Function: value]}>
                                <div className="MuiExpansionPanelSummary-content-44 RSFExpandableSection-summary-6">
                                  <WithStyles(Typography) variant="subheading">
                                    <Typography classes={{...}} variant="subheading" align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                      <h3 className="MuiTypography-root-49 MuiTypography-subheading-56">
                                        Title
                                      </h3>
                                    </Typography>
                                  </WithStyles(Typography)>
                                </div>
                                <WithStyles(IconButton) disabled={false} className="MuiExpansionPanelSummary-expandIcon-45" component="div" tabIndex={-1} aria-hidden="true">
                                  <IconButton classes={{...}} disabled={false} className="MuiExpansionPanelSummary-expandIcon-45" component="div" tabIndex={-1} aria-hidden="true" color="default">
                                    <WithStyles(ButtonBase) className="MuiIconButton-root-73 MuiExpansionPanelSummary-expandIcon-45" centerRipple={true} focusRipple={true} disabled={false} component="div" tabIndex={-1} aria-hidden="true">
                                      <ButtonBase classes={{...}} className="MuiIconButton-root-73 MuiExpansionPanelSummary-expandIcon-45" centerRipple={true} focusRipple={true} disabled={false} component="div" tabIndex={-1} aria-hidden="true" disableRipple={false} type="button">
                                        <div onBlur={[Function: handleEvent]} onFocus={[Function: value]} onKeyDown={[Function: value]} onKeyUp={[Function: value]} onMouseDown={[Function: handleEvent]} onMouseLeave={[Function: handleEvent]} onMouseUp={[Function: handleEvent]} onTouchEnd={[Function: handleEvent]} onTouchMove={[Function: handleEvent]} onTouchStart={[Function: handleEvent]} tabIndex={-1} className="MuiButtonBase-root-46 MuiIconButton-root-73 MuiExpansionPanelSummary-expandIcon-45" role="button" aria-hidden="true">
                                          <span className="MuiIconButton-label-78">
                                            <pure(ExpandMore) className="RSFExpandableSection-expandIcon-8">
                                              <ExpandMore className="RSFExpandableSection-expandIcon-8">
                                                <WithStyles(SvgIcon) className="RSFExpandableSection-expandIcon-8">
                                                  <SvgIcon classes={{...}} className="RSFExpandableSection-expandIcon-8" color="inherit" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-79 RSFExpandableSection-expandIcon-8" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                                                      <g>
                                                        <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                      </g>
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ExpandMore>
                                            </pure(ExpandMore)>
                                          </span>
                                          <WithStyles(TouchRipple) innerRef={[Function: value]} center={true}>
                                            <TouchRipple classes={{...}} center={true}>
                                              <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-85" childFactory={[Function: childFactory]}>
                                                <span className="MuiTouchRipple-root-85" />
                                              </TransitionGroup>
                                            </TouchRipple>
                                          </WithStyles(TouchRipple)>
                                        </div>
                                      </ButtonBase>
                                    </WithStyles(ButtonBase)>
                                  </IconButton>
                                </WithStyles(IconButton)>
                              </div>
                            </ButtonBase>
                          </WithStyles(ButtonBase)>
                        </ExpansionPanelSummary>
                      </WithStyles(ExpansionPanelSummary)>
                      <WithStyles(Collapse) in={false} timeout="auto" aria-hidden="true">
                        <Collapse theme={{...}} classes={{...}} in={false} timeout="auto" aria-hidden="true" collapsedHeight="0px" component="div">
                          <Transition onEnter={[Function: value]} onEntered={[Function: value]} onEntering={[Function: value]} onExit={[Function: value]} onExiting={[Function: value]} addEndListener={[Function: value]} timeout={{...}} in={false} aria-hidden="true" mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onExited={[Function: noop]}>
                            <div className="MuiCollapse-container-92" style={{...}} aria-hidden="true">
                              <div className="MuiCollapse-wrapper-94">
                                <div className="MuiCollapse-wrapperInner-95">
                                  <WithStyles(ExpansionPanelDetails) classes={{...}}>
                                    <ExpansionPanelDetails classes={{...}}>
                                      <div className="MuiExpansionPanelDetails-root-96 RSFExpandableSection-details-5">
                                        Foo
                                      </div>
                                    </ExpansionPanelDetails>
                                  </WithStyles(ExpansionPanelDetails)>
                                </div>
                              </div>
                            </div>
                          </Transition>
                        </Collapse>
                      </WithStyles(Collapse)>
                    </div>
                  </Paper>
                </WithStyles(Paper)>
              </ExpansionPanel>
            </WithStyles(ExpansionPanel)>
          </ExpandableSection>
        </inject-ExpandableSection>
      </WithTheme(inject-ExpandableSection)>
    </WithStyles(WithTheme(inject-ExpandableSection))>
  </MuiThemeProvider>
</Provider>
`;

exports[`ExpandableSection should render AmpExpandableSection when amp == true 1`] = `
<Provider app={{...}}>
  <MuiThemeProvider theme={{...}}>
    <WithStyles(WithTheme(inject-ExpandableSection)) title="Title">
      <WithTheme(inject-ExpandableSection) classes={{...}} title="Title">
        <inject-ExpandableSection theme={{...}} classes={{...}} title="Title">
          <ExpandableSection theme={{...}} classes={{...}} title="Title" amp={true}>
            <WithStyles(AmpExpandableSection) ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title">
              <AmpExpandableSection classes={{...}} ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title" expanded={false}>
                <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                  <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                    <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                  </SideEffect(NullComponent)>
                </HelmetWrapper>
                <amp-accordion disable-session-states={true} class="MoovAmpExpandableSection-accordion-106">
                  <section className="MoovAmpExpandableSection-section-108">
                    <WithStyles(Typography) variant="subheading" component="h3" className="MoovAmpExpandableSection-title-107">
                      <Typography classes={{...}} variant="subheading" component="h3" className="MoovAmpExpandableSection-title-107" align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                        <h3 className="MuiTypography-root-113 MuiTypography-subheading-120 MoovAmpExpandableSection-title-107">
                          Title
                          <pure(ExpandMore) className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-collapse-111">
                            <ExpandMore className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-collapse-111">
                              <WithStyles(SvgIcon) className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-collapse-111">
                                <SvgIcon classes={{...}} className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-collapse-111" color="inherit" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-137 MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-collapse-111" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                                    <g>
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                    </g>
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </ExpandMore>
                          </pure(ExpandMore)>
                          <pure(ExpandMore) className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-expand-110">
                            <ExpandMore className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-expand-110">
                              <WithStyles(SvgIcon) className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-expand-110">
                                <SvgIcon classes={{...}} className="MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-expand-110" color="inherit" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-137 MoovAmpExpandableSection-toggle-109 MoovAmpExpandableSection-expand-110" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                                    <g>
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                    </g>
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </ExpandMore>
                          </pure(ExpandMore)>
                        </h3>
                      </Typography>
                    </WithStyles(Typography)>
                    <div className="MoovAmpExpandableSection-body-112">
                      Foo
                    </div>
                  </section>
                </amp-accordion>
              </AmpExpandableSection>
            </WithStyles(AmpExpandableSection)>
          </ExpandableSection>
        </inject-ExpandableSection>
      </WithTheme(inject-ExpandableSection)>
    </WithStyles(WithTheme(inject-ExpandableSection))>
  </MuiThemeProvider>
</Provider>
`;

exports[`ExpandableSection should render expanded when expanded=true 1`] = `
<Provider app={{...}}>
  <MuiThemeProvider theme={{...}}>
    <WithStyles(WithTheme(inject-ExpandableSection)) title="Title" expanded={true}>
      <WithTheme(inject-ExpandableSection) classes={{...}} title="Title" expanded={true}>
        <inject-ExpandableSection theme={{...}} classes={{...}} title="Title" expanded={true}>
          <ExpandableSection theme={{...}} classes={{...}} title="Title" expanded={true} amp={true}>
            <WithStyles(AmpExpandableSection) ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title">
              <AmpExpandableSection classes={{...}} ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title" expanded={false}>
                <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                  <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                    <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                  </SideEffect(NullComponent)>
                </HelmetWrapper>
                <amp-accordion disable-session-states={true} class="MoovAmpExpandableSection-accordion-152">
                  <section className="MoovAmpExpandableSection-section-154">
                    <WithStyles(Typography) variant="subheading" component="h3" className="MoovAmpExpandableSection-title-153">
                      <Typography classes={{...}} variant="subheading" component="h3" className="MoovAmpExpandableSection-title-153" align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                        <h3 className="MuiTypography-root-159 MuiTypography-subheading-166 MoovAmpExpandableSection-title-153">
                          Title
                          <pure(ExpandMore) className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-collapse-157">
                            <ExpandMore className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-collapse-157">
                              <WithStyles(SvgIcon) className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-collapse-157">
                                <SvgIcon classes={{...}} className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-collapse-157" color="inherit" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-183 MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-collapse-157" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                                    <g>
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                    </g>
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </ExpandMore>
                          </pure(ExpandMore)>
                          <pure(ExpandMore) className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-expand-156">
                            <ExpandMore className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-expand-156">
                              <WithStyles(SvgIcon) className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-expand-156">
                                <SvgIcon classes={{...}} className="MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-expand-156" color="inherit" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-183 MoovAmpExpandableSection-toggle-155 MoovAmpExpandableSection-expand-156" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                                    <g>
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                    </g>
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </ExpandMore>
                          </pure(ExpandMore)>
                        </h3>
                      </Typography>
                    </WithStyles(Typography)>
                    <div className="MoovAmpExpandableSection-body-158">
                      Foo
                    </div>
                  </section>
                </amp-accordion>
              </AmpExpandableSection>
            </WithStyles(AmpExpandableSection)>
          </ExpandableSection>
        </inject-ExpandableSection>
      </WithTheme(inject-ExpandableSection)>
    </WithStyles(WithTheme(inject-ExpandableSection))>
  </MuiThemeProvider>
</Provider>
`;
